os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_peripheral/ble_tizenrt_server.c

```diff
@@ -804,9 +620,54 @@ int rtw_ble_server_set_adv_interval(unsigned int interval)
     } while(new_state.gap_adv_state != GAP_ADV_STATE_IDLE);
 
     debug_print("\r\n[%s] ADV STATE : IDLE", __FUNCTION__);
+}
+
+int rtw_ble_server_set_adv_interval(unsigned int interval)
+{
+    rtw_ble_server_adv_into_idle();
+
     uint16_t adv_int = interval;
     le_adv_set_param(GAP_PARAM_ADV_INTERVAL_MIN, sizeof(adv_int), &adv_int);
     le_adv_set_param(GAP_PARAM_ADV_INTERVAL_MAX, sizeof(adv_int), &adv_int);
     return TRBLE_SUCCESS;
 }
+
+int rtw_ble_server_set_adv_type(trble_adv_type_e type, void *direct_addr)
+{
+    if(type == TRBLE_ADV_TYPE_DIRECT && direct_addr == NULL)
+    {
+        debug_print("\r\n[%s] Invalid Input", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+
+    T_TIZENRT_ADV_TYPE_PARAM *param = os_mem_alloc(0, sizeof(T_TIZENRT_ADV_TYPE_PARAM));
+    if(param == NULL)
+    {
+        debug_print("\n[%s] Memory allocation failed", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    param->adv_type = type;
+    if(type == TRBLE_ADV_TYPE_DIRECT)
+    {
+        param->addr_type = GAP_REMOTE_ADDR_LE_PUBLIC;
+        memcpy(param->adv_direct_addr, direct_addr, TRBLE_BD_ADDR_MAX_LEN);
+    }
+
+    rtw_ble_server_adv_into_idle();
+
+    if(ble_tizenrt_server_send_msg(BLE_TIZENRT_MSG_SET_ADV_TYPE, param) == false)
+    {
+        debug_print("\r\n[%s] msg send fail", __FUNCTION__);
```

Comment:
Is it okay not to free the `param` which allocated from line 643?

Reply:
This modification is unneccessary and I have chang it back.